client/components/command-palette/index.tsx

```diff
@@ -229,7 +231,12 @@ const CommandPalette = () => {
 						) }
 						<CommandInput
 							search={ search }
-							setSearch={ setSearch }
+							setSearch={ ( value ) => {
+								setSearch( value );
+								if ( ! selectedCommandName ) {
+									setLastEnteredSearch( value );
+								}
```

Comment:
I found a corner case  where it doesn't work as expected.
A possible solution could be moving the `setLastEnteredSearch` when a root command is selected and not when the input is updated.


https://github.com/Automattic/wp-calypso/assets/779993/98eda7cf-5c2f-4cee-bc6a-44ff01f3d73d


Steps to reproduce:
* Open the command palette
* Search `dashboa`
* Select the command
* Press `esc` to go back
* Select the command again
* Press `esc` to go back again
* Observe there is no last search state saved, because we didn't updated the command.